client/src/routes/modals/CompletedSessionsModal/CompletedSessionsModal.tsx

```diff
@@ -0,0 +1,226 @@
+import useUserEvents from '../../../lib/user/hooks/useUserEvents';
+import FilterStatus from '../../screens/Journey/components/FilterStatus';
+import {
+  ThumbsDownWithoutPadding,
+  ThumbsUpWithoutPadding,
+} from '../SessionFeedbackModal/components/Thumbs';
+
+import {Spacer16, Spacer32} from '../../../lib/components/Spacers/Spacer';
+import {
+  CommunityIcon,
+  FriendsIcon,
+  MeIcon,
+} from '../../../lib/components/Icons';
+import {
+  SessionMode,
+  SessionType,
+} from '../../../../../shared/src/types/Session';
+
+const Row = styled(Gutters)({
+  flexDirection: 'row',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+});
+
+const CompletedSessionsModal = () => {
+  const {
+    params: {filterSetting},
+  } = useRoute<RouteProp<ModalStackProps, 'CompletedSessionsModal'>>();
+  const {t} = useTranslation('Modal.CompletedSessions');
+  const {completedSessions} = useCompletedSessions();
+  const {feedbackEvents} = useUserEvents();
+  const [selectedMode, setSelectedMode] = useState<
+    SessionMode.async | SessionType.public | SessionType.private
+  >();
+  const [selectedFeedback, setSelectedFeedback] = useState<boolean>();
+
+  const [positiveFeedbacks, negativeFeedbacks] = useMemo(
+    () => partition(f => f.payload.answer, feedbackEvents),
+    [feedbackEvents],
+  );
+
+  const {live: liveSessions, async: asyncSessions} = useMemo(
+    () => groupBy(s => s.payload.mode, completedSessions),
+    [completedSessions],
+  );
+
+  const {private: privateSessions, public: publicSessions} = useMemo(
+    () => groupBy(s => s.payload.type, liveSessions ?? []),
+    [liveSessions],
+  );
+
+  const renderItem = useCallback<ListRenderItem<CompletedSessionEvent>>(
+    ({item}) => (
+      <Gutters>
+        <JourneyNode completedSessionEvent={item} />
+      </Gutters>
+    ),
+    [],
+  );
+
+  const data = useMemo(() => {
+    if (filterSetting === 'feedback') {
+      return feedbackEvents
+        .filter(({payload}) =>
+          selectedFeedback !== undefined
+            ? payload.answer === selectedFeedback
+            : true,
+        )
+        .map(feedbackEvent =>
+          completedSessions.find(
+            completedSessionEvent =>
+              completedSessionEvent.payload.id ===
+              feedbackEvent.payload.sessionId,
+          ),
+        )
+        .filter(Boolean) as CompletedSessionEvent[];
+    }
+
+    if (filterSetting === 'mode') {
+      return completedSessions.filter(({payload}) => {
+        if (selectedMode) {
+          if (selectedMode === SessionMode.async) {
+            return payload.mode === selectedMode;
+          } else {
+            return payload.type === selectedMode;
+          }
+        } else {
+          return true;
+        }
+      });
+    }
+
+    return completedSessions;
+  }, [
+    completedSessions,
+    filterSetting,
+    feedbackEvents,
+    selectedFeedback,
+    selectedMode,
+  ]);
```

Comment:
How about creating a hook out of this so we can test it and reduce the complexity in the component?